Toy DB filling and Twitter unit tests

yuxshao opened the pull request:
1. `.gitignore` was not ignoring new, untracked files. I `git rm --cached -r .` and `git add .` to reset the git index, and changed `db_settings.py` so it tries to take settings from a file `my_db_settings.py`, or else just loads a test config.
2. `generate_db.py` copies 5 Businesses and everything associated with them into a toy database. It also copies 5 tweets + their documents.
3. For that to work, `Twython` was added to `requirements.txt` and `travis_requirements.txt`.
4. `test_twitter_search.py` tests all `twitter_search` functions using `pytest`.
5. Various few small changes (typos, conditions for null arguments)

Let me know when it's good to merge. I'll squash the commits beforehand.

yuxshao committed:
basic toy database with businesses

yuxshao committed:
basic toy database with businesses

yuxshao committed:
basic toy database with businesses

yuxshao committed:
gitignore ignores my_db_settings.py

yuxshao committed:
gitignore ignores my_db_settings.py

yuxshao committed:
rm --cached everything and readded so gitignore would work

yuxshao committed:
rm --cached everything and readded so gitignore would work

yuxshao committed:
gitignore and dbsettings should be okay now? idk

yuxshao committed:
gitignore and dbsettings should be okay now? idk

yuxshao committed:
adds businesses and their locations to testdb

yuxshao committed:
adds businesses and their locations to testdb

yuxshao committed:
Merge branch 'model-classifier-object' of https://github.com/teffland/FoodborneNYC into model-classifier-object

yuxshao committed:
removed old testdb file

teffland commented:
When you have the whole local db working and a couple unit tests that use it, I think that'll be the appropriate time to merge to master

yuxshao committed:
typo in businesses.py?

yuxshao committed:
typo in businesses.py?

yuxshao committed:
working toy db copy

yuxshao committed:
working toy db copy

yuxshao committed:
YelpCategory also copied

yuxshao committed:
YelpCategory also copied

yuxshao committed:
copies everything but tweets over

yuxshao committed:
copies everything but tweets over

yuxshao commented:
Alright, no unit tests have been made yet, but I just wanted to give an update. At this point, everything but tweets now gets copied over, using as much SQLAlchemy as possible. 

I understand you gave an alternate solution on Friday using the JSON directly, but I felt like the error I was showing you probably warranted a deeper look since the error was produced by the line that tried to _retrieve Yelp reviews_ (error was something complaining about the fact that no association between the tables had been set up).

I did some investigation/testing and (after like an hour) noticed that this only happened when the script tried to populate tables after dropping and recreating them in the same session, so if I reset the tables, then closed IPython, then populated the tables, everything would be fine. I think this is because dropping/resetting the tables messes with the global Metadata object, thus breaking things like retrieving data--even from a different DB--despite our using different engines for the two actions. 

Anyway, copying stuff over all works now, and I think I have a better grasp of some of the quirks of SQLAlchemy too. Tweets shouldn't be hard to add.

yuxshao committed:
Merge branch 'model-classifier-object' of https://github.com/teffland/FoodborneNYC into model-classifier-object

yuxshao committed:
clarified test db comment

yuxshao committed:
doesn't break upon reading an emoji in tweet

yuxshao committed:
adds 5 tweets to test db

yuxshao committed:
test sources.twitter_search.make_query

yuxshao commented:
Put tweets in test db. Also, oops, I can't read; used unittest and mock to make a basic test instead of pytests. will revert later today.

teffland commented:
Ok, currently it breaks the integration tests. Let me know when it's fully ready to merge.

yuxshao committed:
twython to requirements

yuxshao commented:
Hey Tom, I took a look at that last night. It's just the test I added that failed, and it looks like it's because Twython's not installed on travis. I added a twython line to `travis_requirements.txt` (and to  `requirements.txt` too, since it wasn't there?). Hopefully that'll get things to work. Let me know if I shouldn't be doing this (if it causes bloat, etc.)!

yuxshao committed:
fixed travis_requirements typo

yuxshao committed:
db_settings outsources to my_db_settings if available

yuxshao committed:
changed db_settings status to make tests work

yuxshao commented:
ahahaha yes finally! basic twitter_query unit test passes. I edited the first post to describe the changes in this branch. I think it's better to merge this branch early--if there aren't any problems--rather than to put a few more changes in here so that whatever next thing to add can branch off master. After all, the test db gets generated--that's a self-contained change in itself.

yuxshao committed:
ignore pytest .cache folder

yuxshao committed:
more flexible Tweet constructor + typo in twitter_search.py

yuxshao committed:
test for tweets_to_Tweets

yuxshao committed:
line length in twitter_search.py reduced to 80

yuxshao committed:
generate_db.clear_tables

yuxshao committed:
typo in twitter_search.py

yuxshao committed:
pytests for all twitter functions

teffland commented on foodbornenyc/sources/twitter_search.py:
```diff
@@ -98,8 +99,9 @@ def query_twitter(how_long=0):
             continue
-        logger.info("%i Total unique tweets in %i:%i:%i time", len(id_set), *sec_to_hms(time.time()-start))
-        new_tweets = [ tweet for tweet in tweets if tweet['id_str'] not in id_set ]
+        logger.info("%i Total unique tweets in %i:%i:%i time", len(id_set),\
```
You don't need the '\' if the lines are wrapped in a function call

teffland commented on foodbornenyc/db_settings.py:
```diff
@@ -1,16 +1,20 @@
-    'access_token_secret': ''
-}
+try:
+    from foodbornenyc.my_db_settings import database_config, twitter_config
```
This seems like a great solution

teffland commented on foodbornenyc/test/generate_db.py:
```diff
@@ -0,0 +1,88 @@
+"""
```
I think we should name this file `generate_test_db`

teffland commented:
It all looks good, I'm going to merge it now.  But just for clarity in your next commit please change `generate_db.py` to `generate_test_db.py`

yuxshao commented:
Great, thanks Tom. I'll address your comments right now in a new branch (though would it have been better to fix them before merging?). The other thing is that before merging it's good to squash the commits in this monster of a branch so that the master branch doesn't get polluted with all these 50+ microcommits. Though... I don't think it's possible to go back and clean the history (for me at least) since it's already merged to the write-protected master.
